diff --git a/libswscale/swscale.c b/libswscale/swscale.c
index 733f57b..85921de 100644
--- a/libswscale/swscale.c
+++ b/libswscale/swscale.c
@@ -269,12 +269,15 @@ yuv2yuvX10_c_template(const int16_t *lumFilter, const int16_t **lumSrc,
              *aDest = CONFIG_SWSCALE_ALPHA ? dest[3] : NULL;
     int shift = 11 + 16 - output_bits - 1;
 
-#define output_pixel(pos, val) \
+#define output_pixel(pos, val) { \
+    int x = output_bits == 9 ? av_clip_uintp2(val >> shift, 9) : \
+                               av_clip_uintp2(val >> shift, 10); \
     if (big_endian) { \
-        AV_WB16(pos, av_clip_uintp2(val >> shift, output_bits)); \
+        AV_WB16(pos, x); \
     } else { \
-        AV_WL16(pos, av_clip_uintp2(val >> shift, output_bits)); \
-    }
+        AV_WL16(pos, x); \
+    } \
+}
     for (i = 0; i < dstW; i++) {
         int val = 1 << (26-output_bits - 1);
         int j;
